pkgs/os-specific/serenity/libc.nix

```diff
@@ -0,0 +1,105 @@
+{ lib
+, stdenv
+, callPackage
+, runCommand
+, pkgsBuildBuild
+, buildPackages
+}:
+
+let
+  inherit (callPackage ./common.nix { }) src version;
+
+  cpuArch = stdenv.targetPlatform.parsed.cpu.name;
+in
+stdenv.mkDerivation rec {
+  inherit src version;
+  pname = "serenitylibc";
+
+  postPatch = ''
+    substituteInPlace Toolchain/CMake/GNUToolchain.txt.in \
+      --replace "gcc-ar" "ar" \
+      --replace "gcc-nm" "nm" \
+      --replace "gcc-ranlib" "ranlib"
+    echo "" > Userland/Libraries/LibC/stubs.cpp
+    echo "" > Tests/LibELF/CMakeLists.txt
+  '';
+
+  depsBuildBuild =
+    [
+      buildPackages.stdenv.cc
+      pkgsBuildBuild.cmake
+      pkgsBuildBuild.ninja
```

Comment:
Shouldn't this be pkgsBuildHost?  (And why not using splicing?)

Reply:
`pkgsBuildHost` causes infinite recursion. idk

Reply:
This happens because `stdenv.cc` on Serenity depends on (evaluating) `openssl` for Serenity, and `openssl.separateDebugInfo` depends on `stdenv.cc`.  But `separateDebugInfo` is forced to Linux-only in make-derivation.nix, so breaking the cycle would be easy here — just swap that conditional so that `stdenv.hostPlatform.isLinux` is first, and then it won't need to try evaluating the value at all for non-Linux.